### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by vpuzakov**
**Frontend**
- Pull request: https://github.com/phoenix61/nodejs-aws-fe/pull/2
- CloudFront: https://d1m1vr3bvrn85w.cloudfront.net/

**Endpoints**
- GET - https://l2zh9lwlf4.execute-api.eu-west-1.amazonaws.com/dev/products
- GET - https://l2zh9lwlf4.execute-api.eu-west-1.amazonaws.com/dev/products/7567ec4b-b10c-48c5-9345-fc73c48a80aa
- Not found - https://l2zh9lwlf4.execute-api.eu-west-1.amazonaws.com/dev/products/1

**What was done**
- productList - **done**
- productById - **done**
- async/await is used in lambda functions - **done**
- ES6 modules are used for product-service implementation - **done**
- Webpack is configured for product-service - **done**
- SWAGGER documentation is created for product-service - **NOT done** 
- Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered) (You may use JEST) - **NOT done**
- Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase - **done**
- Main error scenarios are handled by API ("Product not found" error, try-catch blocks are used in lambda handlers) - **done**

### CODE REVIEW

File: product-service/src/utils/responseBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { APIGatewayProxyResult } from 'aws-lambda';
+
+const corsHeaders = {
+  'Access-Control-Allow-Methods': '*',
+  'Access-Control-Allow-Headers': '*',
+  'Access-Control-Allow-Origin': '*'
+};
+
+const successResponse = (body: Object, statusCode: number = 200): APIGatewayProxyResult => {
```
Comment (kathrynowy) [inline product-service/src/utils/responseBuilder.ts:9]: i believe, that 200 code as default is not a very good idea

**Review by kathrynowy — APPROVED** [review]

**Comment by kathrynowy** [comment]
getAllProducts is not an async function
so, async await weren't used in lambda handlers

File: product-service/src/utils/responseBuilder.ts (hunk 1 above)
Comment (vpuzakov) [reply product-service/src/utils/responseBuilder.ts:9]: Why not? It's a successful response.

**Comment by vpuzakov** [comment]
@kathrynowy  Agree, now they don't. But they will :)
Otherwise, there is no place in my Lambda implementation where I can use async/await

**Comment by kathrynowy** [comment]
@phoenix61, you can add any async call to open API or create a new weather info lambda, as was proposed by the mentors in the video )

**Comment by kathrynowy** [comment]
or you can mock promise, the most straightforward solution

**Pull request merged**